### PR TITLE
#1737 Convert to Picture: Unsupported Object

### LINKED ISSUE #1737: Convert to Picture: Unsupported Object
## Bug Report  
image

#### Environment (e.g. version of Microsoft PowerPoint)
Powerpoint 2016, WIndows 10
#### Steps to replicate
1. Insert any shape (rectangle in this example)
2. Open shapes lab.
3. Add shape to shapes lab.
#### Expected behavior
No error message.
#### Actual behavior
Error message: "Convert to Picture only supports Shapes and Charts."
#### Additional information
`Convert to picture` works fine on the above tested shapes.
Error seems to be caused by `ConvertAndSave` under `ConvertToPicture.cs`.

Edit: The line `if (!ShapeUtil.IsSelectionShapeOrText(selection))` from `CustomShapePane.cs` conflicts with `if (ShapeUtil.IsShapeRangeShapeOrText(selectedShapes))` from `ConvertToPicture.cs`. This seems to be the cause.

### PR DESCRIPTION
**Pull request by YuPeiHenry**
Fixes #1737

Ready for review (test cases included). Tested on Powerpoint 2016.

**Outline of Solution**
As mentioned in issue, the line `if (!ShapeUtil.IsSelectionShapeOrText(selection))` from `CustomShapePane.cs` conflicts with `if (ShapeUtil.IsShapeRangeShapeOrText(selectedShapes))` from `ConvertToPicture.cs`. This seems to be the cause. Both now use `if (!ShapeUtil.IsShapeRangeShapeOrText(...))`.

Added tests to confirm that the error pops up only when adding placeholder objects, in `ShapesLab FT`: TestSavePlaceholderToShapesLabWithAddShapesButton.
Added functions to MessageBoxUtil.cs: `ExpectMessageBoxWillNotPopUp`

### CODE REVIEW

File: PowerPointLabs/Test/FunctionalTest/ShapesLabTest.cs [hunk 1]
```diff
@@ -15,14 +17,15 @@ public class ShapesLabTest : BaseFunctionalTest
         private const int AddShapesShapesSlide = 6;
         private const int AddShapesTestSlide = 7;
         private const int AddShapesExpSlide = 8;
+        private const int AddShapesPlaceholder = 9;
```
Comment (ChesterSng) [inline PowerPointLabs/Test/FunctionalTest/ShapesLabTest.cs]: Hi @YuPeiHenry, can you rename this to `AddShapesPlaceholderSlide` instead for consistency?

**Comment by blewjy** [comment]
Tested and working on PPT2013 💯

File: PowerPointLabs/Test/FunctionalTest/ShapesLabTest.cs (hunk 1 above)
Comment (YuPeiHenry) [reply PowerPointLabs/Test/FunctionalTest/ShapesLabTest.cs]: @ChesterSng Thanks for the correction! I have changed the naming to be consistent.

**Comment by YuPeiHenry** [comment]
@ChesterSng I missed out the description in the slides for the new slide I added in ShapesLab slide, and added it in the recent commit.

File: PowerPointLabs/Test/Util/MessageBoxUtil.cs [hunk 2]
```diff
@@ -65,6 +73,39 @@ class MessageBoxUtil
...
+        {
+            // MessageBox in pptlabs will block the whole thread,
+            // so multi-thread is needed here.
+            Task taskToVerify = new Task(() =>
+            {
+                // try to find messagebox window
+                IntPtr msgBoxHandle = IntPtr.Zero;
+                while (msgBoxHandle == IntPtr.Zero && retryCount > 0)
+                {
+                    msgBoxHandle = NativeUtil.FindWindow("#32770", title);
```
[the remark this answers is not in the document]
Comment (YuPeiHenry) [reply PowerPointLabs/Test/Util/MessageBoxUtil.cs]: @leeyh20 It came from another existing function in MessageBoxUtil. I will assign it to a constant then.

**Comment by YuPeiHenry** [comment]
@leeyh20 I have moved the magic number for message box identifier into a constant.

**Comment by leeyh20** [comment]
@YuPeiHenry Could you check why the AppVeyor build is failing?

**Comment by YuPeiHenry** [comment]
@leeyh20 `ERROR: Error 404 on https://sonarqube.com/api/sources/hash?key=powerpointlabs%3Adev-release%3APowerPointLabs%2FShapesLab%2FCustomShapePane.cs : {"errors":[{"msg":"Component key 
'powerpointlabs:dev-release:PowerPointLabs/ShapesLab/CustomShapePane.cs' not fo
und"}]}`

Is this a sonarqube error? The other recent commits are getting similar errors.

**Comment by leeyh20** [comment]
@YuPeiHenry   

> FunctionalTest\ShapesLabTest.cs(60,74): error CS0103: The name 'AddShapesPlaceholder' does not exist in the current context [C:\projects\powerpointlabs\PowerPointLabs\Test\Test.csproj]
>     742 Warning(s)
>     1 Error(s)
> Time Elapsed 00:02:05.01
> 

I found this error in the AppVeyor build log. Do you get this error when you run the test?

**Comment by YuPeiHenry** [comment]
@leeyh20 Thank you for your guidance. A bad mistake on my part (I should have used refactor instead of just renaming from `AddShapesPlaceholder` to `AddShapesPlaceholderSlide`.)

**Comment by YuPeiHenry** [comment]
Does this line `WARNING: WARN: Failed to set working directory hidden: C:\projects\powerpointlabs\.sonarqube\out\.sonar` have any significance? Or is this a consequence of a mistake in code?

**Pull request closed without merging**